tests/Fixer/PhpUnit/PhpUnitAttributesFixerTest.php

```diff
@@ -699,6 +699,31 @@ public static function provideFooCases() { yield [true]; yield [false]; }
                 'keep_annotations' => true,
             ],
         ];
+
+        yield 'data provider with trailing parentheses' => [
+            <<<'PHP'
+                <?php
+                class TheTest extends \PHPUnit\Framework\TestCase {
+                    /**
+                     */
+                    #[\PHPUnit\Framework\Attributes\DataProvider('provideFooCases()')]
+                    #[\PHPUnit\Framework\Attributes\DataProviderExternal(AnotherTest::class, 'provideBarCases()')]
```

Comment:
Shouldn't it be like this?

```suggestion
                    #[\PHPUnit\Framework\Attributes\DataProvider('provideFooCases')]
                    #[\PHPUnit\Framework\Attributes\DataProviderExternal(AnotherTest::class, 'provideBarCases')]
```

(in order to fail pipeline and show that it does not work as expected right now).

Reply:
Then maybe there should be 3 commits:
- test case with existing, invalid behaviour (CI green)
- changes in expectation (CI red)
- actual fix (CI green)

First 2 commits could be pushed together IMHO, as most interesting is to show it does not work as expected (first commit would be mostly for historical reasons). Then the 3rd commit provides a solution. This way it would either show wrong behaviour (1st commit) and correct behaviour (2nd and 3rd commit).

But it's up to you.